TechnicalDocuments/SnapshotTesting.md

```diff
@@ -0,0 +1,50 @@
+Improved Snapshot Testing
+=========================
+
+## Introduction
+
+This proposal is to replace the library that we use for snapshot testing ([`iOSSnapshotTestCase`](https://github.com/uber/ios-snapshot-test-case/)) with [`SnapshotTesting`](https://github.com/pointfreeco/swift-snapshot-testing).
+
+## Motivation
+
+### Cons of `iOSSnapshotTestCase`
+* `iOSSnapshotTestCase` is an old library written in Objective-C that can only snapshot a `UIView` or `CALayer`.
+* It has on occasion seemed to be quite fragile, reporting diffs when nothing changed, to the extent that we have had to increase the failure tolerance to quite a high value, and have even on a couple of occasions considered dropping snapshot testing altogether.
+* it stores all references images in a single folder for all tests
+* you have to manually find the failure images and diffs to see what went wrong (or use an app like FBSnapshotsViewer)
+* You always have to explicityly set `recordMode` to `true` even if the test has no reference images yet.
+
+### Pros of `SnapshotTesting`
+* `SnapshotTesting` is a brand new library written in Swift by the [Pointfree](pointfree.co) duo. It's a great example of a protocol based API using struct based protocol witnesses as explained in the Pointfree videos on the subject.
+* It supports snapshotting anything, not just `UIView` or `CALayer`. For example, you could snapshot a `URLRequest` instead of having to laboriously type out the expected values by hand.
+You can snapshot anything as a text representation, for example, a view hierarchy, a Bento render tree, a struct, or a JSON dictionary.
+* It is easy to extend the library to add explicit support for snapshotting of other types, instead of just using the `Any` strategy.
+* Snapshots are stored in a `__Snapshots__` folder in the same folder hierarchy as the test source. This makes it easy to find relevant snapshots.
+* It uses `XCTAttachment`s to store failure snapshots and diffs, which makes these available directly from the Xcode test report navigator, without requiring any other software or needing to hunt for them.
+* It is fully Linux compatiable. Allowing snapshot testing to be used on projects that are not iOS or macOS (eg our merge bot).
+* Even if `record` is set to `false` it will still record reference snapshots if none currently exist, so you only have to set it to `true` when updating snapshots becuase of changes.
+
+### Cons of `SnapshotTesting`
+* It's a very new library and in active development. The developers themselves state *"This library should be considered alpha, and not stable. Breaking changes will happen often."* However, provided we stick to a specific version or commit in the Podfile, this should not be a big issue.
```

Comment:
This changes today 😄

https://www.pointfree.co/blog/posts/23-snapshottesting-1-0-delightful-swift-snapshot-testing